core/rawdb/ancient_scheme.go

```diff
@@ -44,10 +46,36 @@ var chainFreezerNoSnappy = map[string]bool{
 	ChainFreezerDifficultyTable: true,
 }
 
+const (
+	// stateHistoryTableSize defines the maximum size of freezer data files.
+	stateHistoryTableSize = 2 * 1000 * 1000 * 1000
+
+	// stateHistoryAccountIndex indicates the name of the freezer state history table.
+	stateHistoryMeta         = "history.meta"
+	stateHistoryAccountIndex = "account.index"
+	stateHistoryStorageIndex = "storage.index"
+	stateHistoryAccountData  = "account.data"
+	stateHistoryStorageData  = "storage.data"
+)
+
+var stateHistoryFreezerNoSnappy = map[string]bool{
+	stateHistoryMeta:         true,
+	stateHistoryAccountIndex: false,
+	stateHistoryStorageIndex: false,
+	stateHistoryAccountData:  false,
+	stateHistoryStorageData:  false,
+}
```

Comment:
Do we ever need only one part of a history object but not another? I.e. does it make sense to split the history across 5 freezer tables instead of just having one?

Reply:
Replied above.

Merging tables together into a single one will increase the complexity.
For example, I will offset to indicate the location of an account in account
data table, If we merge five tables together, the logic for computing the
offset will be more complicated.

And also, when serving the archive state request, only a part of history
is needed. I think it makes sense to separate these five to different tables.

Reply:
Ah, looking up stuff in archive mode and just accessing bits and bobs makes sense! :+1: